code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1794,6 +1825,20 @@ pub(crate) fn claim_of_stake<T: Config>(
 }
 
 impl<T: Config> Pallet<T> {
+	pub fn register_protocol_asset(nonce: u64) -> Result<T::AssetId, DispatchError> {
+		T::AssetsTransactor::create_local_asset(
+			T::PalletId::get().0,
```

Comment:
can just use register<T:PalletInfo> and share method with ALL pallets. So you hide 1. pallet id 2. on counter, with pallet id prefix. So final trait would not have any stuff, and will be just register asset. So having protocol and nonce - do not see any value. Nor in hashig, while prefix with 000number or just number seems work well. How then one will be fetch asset? Where is API to et ALL assets for protocol?

Reply:
What is the use case for getting all of the assets from a protocol?

Reply:
> can just use register<T:PalletInfo> and share method with ALL pallets. So you hide 1. pallet id 2. on counter, with pallet id prefix. So final trait would not have any stuff, and will be just register asset.

The idea was to ensure that the registry was not ridged in its use but also supported convenient workflows for name-spaced assets.

> How then one will be fetch asset? Where is API to et ALL assets for protocol?

Currently, our protocols track their own assets.

The mentioned API was never a requirement, but could be added in the future.

Reply:
> What is the use case for getting all of the assets from a protocol?

yeah. if it is needed. if not - why to have some ranges at all. just to build indexers on top of some scattered storages insitede codebase...